### PR TITLE
Add a dataset downloader job to the Finetuning example

### PR DESCRIPTION
**Pull request by NavarrePratt**
This PR contains a bug fix in the argo workflow yaml and adds a kubernetes job to download a sample dataset to finetune on.

**Bug Fix**
The image that the argo workflow used for finetuning uses a `--fp16` flag to train in lower precision but the workflow was attempting to pass a value to the parameter: `--fp16 true`.
These changes will insert the `--fp16` flag to the container command based on whether or not the corresponding argo workflow parameter is set to `true`.

The first commit on this PR implements a fix by using multiple templates, one for fp16 and one without. The second commit, which I prefer, uses string templating to pass the entire command in as a command argument.

**Dataset Downloader**
Currently, the documentation assumes you already have a dataset ready to use when fine tuning. These changes (third commit) add a dataset downloader kubernetes job that will download book text files available on [SmashWords](https://www.smashwords.com/) which was used to create the original [BookCorpus](https://huggingface.co/datasets/bookcorpus) dataset and further variations (such as a part of [The Pile](https://pile.eleuther.ai/)).

The corresponding documentation changes can be found [here](https://app.gitbook.com/o/-M8EEYiNkaJsT8ISX0kX/s/-M83TghsCfsi8FCYs2DZ/~/changes/ORi5osiMLnnQZazjyzdG/compass/finetuning-machine-learning-models).

### CODE REVIEW

File: finetuner-workflow/dataset-downloader/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,24 @@
+FROM golang:1.19-alpine as source
```
Comment (wbrown) [inline finetuner-workflow/dataset-downloader/Dockerfile:1]: Can we switch the `dataset-downloader/Dockerfile` to using `ko`? You can see an example usage of this: here: https://github.com/wbrown/gpt_bpe/blob/main/.github/workflows/main.yml

File: finetuner-workflow/finetune-workflow.yaml [hunk 2]
```diff
@@ -158,50 +158,33 @@ spec:
...
-          - name: cache
-            value: "/{{workflow.parameters.pvc}}/cache/"
-          - name: torch_cache
-            value: "/{{workflow.parameters.pvc}}/torch/"
-          - name: no_resume
-            value: "{{workflow.parameters.no_resume}}"
-          - name: logs
-            value: "/{{workflow.parameters.pvc}}/{{workflow.parameters.logs}}"
+          - name: finetuner_params
+            # Pass the params in as one string so we can template in the conditional flags
```
Comment (wbrown) [inline finetuner-workflow/finetune-workflow.yaml:162]: Nice workaround!

File: finetuner-workflow/dataset-downloader/Dockerfile (hunk 1 above)
Comment (wbrown) [reply finetuner-workflow/dataset-downloader/Dockerfile:1]: We might want to get gitlab setup to build this/put this in ml-images. @todie, @salanki?

**Review by wbrown — CHANGES_REQUESTED** [review]

File: finetuner-workflow/dataset-downloader/Dockerfile (hunk 1 above)
Comment (NavarrePratt) [reply finetuner-workflow/dataset-downloader/Dockerfile]: Are you suggesting we get a build pipeline set up with this repo or moving the downloader to [the ml containers repo](https://github.com/coreweave/ml-containers) and building it from there?

**Pull request merged**